nemo/lightning/pytorch/strategies/megatron_strategy.py

```diff
@@ -267,6 +267,8 @@ def __init__(
         elif ddp == "pytorch":
             self.ddp_config = None
             self.no_ddp_communication_hook = False
+        elif ddp is None:
+            self.ddp_config = None
```

Comment:
1. @marcromeyn are you OK with this change i.e. making `ddp` optional? (btw. let's also update typing in `MegatronStrategy.__init__` params accordingly)
2. Maybe `ddp = "pytorch"` will do (he previous `elif`)?